clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -888,9 +888,10 @@ public void run() {
                         long now = time.milliseconds();
 
                         if (coordinatorUnknown()) {
-                            if (findCoordinatorFuture == null)
-                                lookupCoordinator();
-                            else
+                            if (findCoordinatorFuture == null) {
+                                if (lookupCoordinator().failed())
```

Comment:
Might be worth a comment explaining that the immediate check ensures that we backoff properly in the case that no brokers are available to connect to. I think these checks could also be condensed:

```java
if (findCoordinatorFuture != null || lookupCoordinator().failed())
  AbstractCoordinator.this.wait(retryBackoffMs);
```